### PR TITLE
infra: fix DeleteAllOf rbac

### PR DESCRIPTION
**Pull request by zirain**
following up: https://github.com/envoyproxy/gateway/pull/4430

e2e tracked in: https://github.com/envoyproxy/gateway/issues/4453

### CODE REVIEW

File: charts/gateway-helm/templates/infra-manager-rbac.yaml [hunk 1]
```diff
@@ -11,10 +11,12 @@ rules:
   resources:
   - serviceaccounts
   - services
+  - configmaps
```
Comment (arkodg) [inline charts/gateway-helm/templates/infra-manager-rbac.yaml:14]: why is this needed ?

Comment (zirain) [reply charts/gateway-helm/templates/infra-manager-rbac.yaml:14]: For RateLimit

Comment (arkodg) [reply charts/gateway-helm/templates/infra-manager-rbac.yaml:14]: dont we use xds for configuring intent ?

Comment (zirain) [reply charts/gateway-helm/templates/infra-manager-rbac.yaml:14]: https://github.com/envoyproxy/gateway/blob/958df4899d826c03ecd3aafaea379289bba85f3a/internal/infrastructure/kubernetes/ratelimit/resource_provider.go#L36

it's for statsd to prometheus configuration.

Comment (arkodg) [reply charts/gateway-helm/templates/infra-manager-rbac.yaml:14]: ah

**Review by arkodg — APPROVED** [review]

**Review by shawnh2 — APPROVED** [review]

**Comment by arkodg** [comment]
should we backport this @zirain  ?

**Comment by zirain** [comment]
both or not
